src/reducers/existingThoughtChange.ts

```diff
@@ -56,18 +56,12 @@ const existingThoughtChange = (state: State, { oldValue, newValue, context, show
   const thoughtsRankedLiveOld = showContexts
     ? contextOf(contextOf(thoughtsRanked)).concat({ value: oldValue, rank: headRank(contextOf(thoughtsRanked)) }).concat(head(thoughtsRanked))
     : contextOf(thoughtsRanked).concat({ value: oldValue, rank })
-
   // find exact thought from thoughtIndex
   const exactThought = thoughtOld.contexts.find(thought => equalArrays(thought.context, context) && thought.rank === rank)
   const id = headId(thoughtsRanked) || exactThought!.id as string
   const archived = exactThought ? exactThought.archived : null
-
-  const cursorNew = state.cursor && state.cursor.map(thought => thought.value === oldValue && thought.rank === (rankInContext || rank)
-    ? { value: newValue, rank: thought.rank }
-    : thought
-  )
-
-  const newPath = thoughtsRanked.slice(0, thoughtsRanked.length - 1).concat({ value: newValue, rank: thoughtsRanked.slice(thoughtsRanked.length - 1)[0].rank })
+  const cursorNew = state.cursor && [...state.cursor.slice(0, -1), { ...head(state.cursor), value: newValue }]
+  const newPath = thoughtsRanked.slice(0, thoughtsRanked.length - 1).concat({ value: newValue, rank: rankInContext || rank })
```

Comment:
Please do not use array methods on `Path` types. They are meant to be ADT's with clear methods of interaction. The underlying array is an implementation detail that consumers should not be aware of.

Use `head` and `contextOf`.